File: features/payment/pms/src/main/java/com/tokopedia/pms/bankdestination/view/adapter/BankListViewHolder.java
package com.tokopedia.pms.bankdestination.view.adapter;

import android.view.View;
import android.widget.TextView;

import com.tokopedia.abstraction.base.view.adapter.viewholders.AbstractViewHolder;
import com.tokopedia.pms.R;
import com.tokopedia.pms.bankdestination.view.model.BankListModel;

/**
 * Created by zulfikarrahman on 7/5/18.
 */

public class BankListViewHolder extends AbstractViewHolder<BankListModel> {
    public static int Layout = R.layout.item_bank_list;

    private TextView bankName;

    public BankListViewHolder(View itemView) {
        super(itemView);
        bankName = itemView.findViewById(R.id.bank_name);
    }

    @Override
    public void bind(BankListModel element) {
        bankName.setText(element.getBankName());
    }
}

File: features/payment/pms/src/main/java/com/tokopedia/pms/bankaccount/view/viewholder/BankListViewHolder.kt
package com.tokopedia.pms.bankaccount.view.viewholder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.tokopedia.pms.R
import com.tokopedia.pms.bankaccount.data.model.BankListModel
import com.tokopedia.pms.databinding.ItemBankListBinding

/**
 * Created by zulfikarrahman on 7/5/18.
 */
class BankListViewHolder(
    view: View,
    val clickAction: (BankListModel) -> Unit
) : RecyclerView.ViewHolder(view) {

    private val binding = ItemBankListBinding.bind(view)

    fun bind(element: BankListModel) {
        binding.bankName.text = element.bankName
        itemView.setOnClickListener { clickAction(element) }
    }

    companion object {
        var Layout = R.layout.item_bank_list
    }
}
